src/ensembl/src/shared/components/instant-download/instant-download-fetch/fetchForTranscript.ts

```diff
@@ -29,51 +34,82 @@ type Options = {
 };
 
 type FetchPayload = {
-  transcriptId: string;
+  genomeId: string;
   geneId: string;
+  transcriptId: string;
   options: Options;
 };
 
 export const fetchForTranscript = async (payload: FetchPayload) => {
   const {
+    genomeId,
     geneId,
     transcriptId,
     options: { transcript: transcriptOptions, gene: geneOptions }
   } = payload;
-  const urls = buildUrlsForTranscript(transcriptId, transcriptOptions);
+  const productGeneratingContext = await fetchSequenceChecksums({
+    genomeId,
+    transcriptId
+  });
+  const urls = buildUrlsForTranscript(
+    { geneId, productGeneratingContext },
+    transcriptOptions
+  );
+
   if (geneOptions.genomicSequence) {
-    urls.push(buildFetchUrl(geneId, 'genomicSequence'));
+    urls.push(buildFetchUrl({ geneId }, 'genomicSequence'));
   }
+
   const sequencePromises = urls.map((url) =>
     fetch(url).then((response) => response.text())
   );
   const sequences = await Promise.all(sequencePromises);
-  const combinedFasta = sequences.join('\n');
+  const combinedFasta = sequences.join('\n\n');
 
   downloadAsFile(combinedFasta, `${transcriptId}.fasta`, {
     type: 'text/x-fasta'
   });
 };
 
 const buildUrlsForTranscript = (
-  id: string,
+  data: {
+    geneId: string;
+    productGeneratingContext: ProductGeneratingContextFragment;
+  },
   options: Partial<TranscriptOptions>
 ) => {
   return options
     ? transcriptOptionsOrder
         .filter((option) => options[option])
-        .map((option) => buildFetchUrl(id, option))
+        .map((option) => buildFetchUrl(data, option))
     : [];
 };
 
-const buildFetchUrl = (id: string, sequenceType: TranscriptOption) => {
-  const sequenceTypeToTypeParam: Record<TranscriptOption, string> = {
+const buildFetchUrl = (
+  data: {
+    geneId: string;
+    productGeneratingContext?: ProductGeneratingContextFragment;
+  },
+  sequenceType: TranscriptOption
+) => {
+  const sequenceTypeToContextType: Record<TranscriptOption, string> = {
     genomicSequence: 'genomic',
-    proteinSequence: 'protein',
+    proteinSequence: 'product',
     cdna: 'cdna',
     cds: 'cds'
   };
-  const typeParam = sequenceTypeToTypeParam[sequenceType];
 
-  return `https://rest.ensembl.org/sequence/id/${id}?content-type=text/x-fasta&type=${typeParam}`;
+  if (sequenceType === 'genomicSequence') {
+    return `https://rest.ensembl.org/sequence/id/${data.geneId}?content-type=text/x-fasta&type=${sequenceTypeToContextType.genomicSequence}`;
+  } else {
+    const contextType = sequenceTypeToContextType[
+      sequenceType
+    ] as keyof ProductGeneratingContextFragment;
+    const sequenceChecksum =
+      data.productGeneratingContext &&
+      data.productGeneratingContext[contextType]?.sequence_checksum;
```

Comment:
Is it safe to assume that the checksum will always be defined?
<img width="659" alt="Screenshot 2021-01-28 at 09 01 51" src="https://user-images.githubusercontent.com/3085815/106114472-7c3ee100-6147-11eb-9d3a-1c8a9d836040.png">

Reply:
> Is it safe to assume that the checksum will always be defined?

We have coding and non-coding transcripts; those will have different sets of checksums.

Reply:
> > Is it safe to assume that the checksum will always be defined?
> 
> We have coding and non-coding transcripts; those will have different sets of checksums.

I assume we don't need to change anything for this then?